pkg/prometheus/promcfg.go

```diff
@@ -227,6 +250,13 @@ func (cg *ConfigGenerator) GenerateConfig(
 	additionalAlertManagerConfigs []byte,
 	ruleConfigMapNames []string,
 ) ([]byte, error) {
+	// Validate Prometheus Config Inputs at Prometheus CRD level
+	err := validateConfigInputs(p)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid config value passed")
+	}
```

Comment:
I think we can avoid wrapping the error, the returned error should already be clear enough.

```suggestion
	if err := validateConfigInputs(p); err != nil {
		return nil, err
	}
```